### PR TITLE
fix(cancelación): borrador FFM huérfano no bloquea cancelar SI

### PR DESCRIPTION
**Pull request by luisrms69**
## Objetivo

Desbloquear la cancelación de una Sales Invoice cuando existe una FFM en BORRADOR
nunca timbrada (duplicado huérfano), sin borrarla ni modificarla. Incidente real de
LlantasCS: ACC-SINV-2026-02932.

## Cambios principales

- `api/fiscal_operations.py`: nuevo helper `_es_ffm_borrador_sin_timbrar(ffm_row, active_ffm_name)`.
  Una FFM con `docstatus=0` + `status=BORRADOR` + sin `fm_uuid` + sin `facturapi_id`,
  que además no sea la FFM activa de la SI (`fm_factura_fiscal_mx`), no es fiscalmente
  activa → no bloquea la cancelación y no se desvincula ni se modifica.
- `cancelar_si_post_fiscal`: aplica el helper en el guard de validación y en la
  desvinculación. El guard preexistente (FFM activa debe estar CANCELADO) queda intacto.
  Cualquier FFM con evidencia fiscal, `docstatus!=0`, status distinto de BORRADOR
  (PROCESANDO/PENDIENTE) o que sea la FFM activa sigue exigiendo CANCELADO.

## Documentación actualizada

- No aplica: bugfix a un caso borde de un guard existente; no cambia el flujo de
  cancelación documentado (`docs/usuario/cancelar-cfdi.md` no describe este guard).
  La integridad de cancelación FFM ya está cubierta por ADR 0036.

## Validaciones realizadas

- Tests: `facturacion_fiscal/tests/test_cancel_guard_draft_ffm.py` — 9/9 OK
  (predicado + integración: draft huérfano no bloquea y queda intacto; draft con UUID,
  PROCESANDO, submitted no-CANCELADO y draft que es la FFM activa siguen bloqueando).
- ruff check OK · bandit sin issues · mkdocs build --strict OK.
- Validación GUI sobre clon de producción (llantascs-v16.dev): SI cancelada
  (docstatus=2), FFM cancelada desvinculada con evidencia intacta, borrador huérfano
  (FFMX-2026-00030) sin tocar.
- Suite completa del app: no re-corrida en este PR (falla preexistente conocida e
  independiente: test_layer2_cross_module_validation).

## Fuera de alcance

- Causa raíz del duplicado (doble creación de FFM en el submit por guard en memoria en
  `_get_or_create_factura_fiscal`).
- Bloqueo de timbrado / estado formal DESCARTADO para borradores duplicados.
- Centralización de la cancelación nativa de ERPNext.

## Riesgos / pendientes

- El duplicado huérfano permanece ligado a la SI cancelada (conserva trazabilidad); su
  limpieza formal queda pendiente y no forma parte de este fix.
- Deploy a producción LlantasCS (bench v15) requiere `bench migrate` + `bench build`
  vía MSP/Ansible.

## Summary by CodeRabbit

* **Bug Fixes**
  * Cancellation now ignores draft, unissued fiscal documents that are not actually active, preventing unnecessary blocking.
  * Draft records are preserved during cancellation and are no longer unlinked or altered unless they are truly active.
  * Improved handling for edge cases where a draft becomes fiscally relevant, ensuring cancellation is still blocked when appropriate.

* **Tests**
  * Added coverage for draft-vs-active cancellation behavior, including linked, processing, and submitted fiscal document scenarios.

### CODE REVIEW

File: facturacion_mexico/facturacion_fiscal/tests/test_cancel_guard_draft_ffm.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+	def test_draft_no_bloquea_y_permanece_intacta(self):
+		# El cancel de ERPNext (GL) se simula: aislamos el guard/desvinculación.
+		with mock.patch("frappe.model.document.Document.cancel") as mocked_cancel:
+			result = cancelar_si_post_fiscal(self.si)
+
+		# La cancelación continuó (guard NO bloqueó) y se llamó al cancel de la SI
+		self.assertTrue(result.get("ok"))
+		self.assertTrue(mocked_cancel.called)
+
```
[the remark this answers is not in the document]
Comment (luisrms69) [reply facturacion_mexico/facturacion_fiscal/tests/test_cancel_guard_draft_ffm.py:182]: Deliberate and kept as-is. The unit under test here is the guard `cancelar_si_post_fiscal` (which FFMs block/are unlinked), not ERPNext's cancellation machinery. Exercising the real `Document.cancel()` would require a fully valid submitted Sales Invoice with GL accounts/company setup, turning a focused guard unit test into a heavy ERPNext-integration test that validates ERPNext, not the changed code.

The real end-to-end cancellation (docstatus transition, GL reversal, FFM unlink, orphan draft left intact) was validated in the GUI on a production clone (`llantascs-v16.dev`) for the actual incident SI `ACC-SINV-2026-02932`. So the real `cancel()` path is covered — just not inside this unit test.

**Review by it-consultoria-negocios — APPROVED** [review]
